src/nvhttp.cpp

```diff
@@ -573,6 +574,30 @@ namespace nvhttp {
       return false;
     }
 
+    // ensure pin is 4 digits
+    if (pin.size() != 4) {
+      tree.put("root.paired", 0);
+      tree.put("root.<xmlattr>.status_code", 400);
+      tree.put(
+        "root.<xmlattr>.status_message", "Pin must be 4 digits, " + std::to_string(pin.size()) + " provided");
+      return false;
+    }
+
+    // ensure all pin characters are numeric
+    if (!std::all_of(pin.begin(), pin.end(), ::isdigit)) {
+      tree.put("root.paired", 0);
+      tree.put("root.<xmlattr>.status_code", 400);
+      tree.put("root.<xmlattr>.status_message", "Pin must be numeric");
+      return false;
+    }
+
+    if (std::all_of(pin.begin(), pin.end(), [&](char c) { return c == pin[0]; })) {
```

Comment:
Let's remove this, since these are still valid PINs can happen due to random chance.

If someone has gone far enough to make custom Moonlight builds with a fixed PIN, they could also rebuild Sunshine without this check or use a similarly simple PIN like 1234.